src/sys/windows_impl.rs

```diff
@@ -291,3 +293,77 @@ fn round_up(num_to_round: i64, multiple: i64) -> i64 {
     debug_assert_eq!((multiple & (multiple - 1)), 0);
     (num_to_round + multiple - 1) & -multiple
 }
+
+/// Checks whether reflink is supported on the filesystem for the specified source and target paths.
+///
+/// This function verifies that both paths are on the same volume and that the filesystem supports
+/// reflink.
+pub fn check_reflink_support(
+    from: impl AsRef<Path>,
+    to: impl AsRef<Path>,
+) -> io::Result<ReflinkSupport> {
+    let from_volume = get_volume_path(from)?;
+    let to_volume = get_volume_path(to)?;
+
+    let from_guid = get_volume_guid_path(&from_volume)?;
+    let to_guid = get_volume_guid_path(&to_volume)?;
+    if from_guid != to_guid {
+        // The source and destination files must be on the same volume
+        return Ok(ReflinkSupport::NotSupported);
+    }
+
+    let volume_flags = get_volume_flags(&from_volume)?;
+    if volume_flags & FILE_SUPPORTS_BLOCK_REFCOUNTING == FILE_SUPPORTS_BLOCK_REFCOUNTING {
+        Ok(ReflinkSupport::Supported)
+    } else {
+        Ok(ReflinkSupport::NotSupported)
+    }
+}
+
+/// A wrapper function for
+/// [GetVolumePathNameW](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getvolumepathnamew)
+/// that retrieves the volume mount point where the specified path is mounted.
+fn get_volume_path(path: impl AsRef<Path>) -> io::Result<Vec<u16>> {
+    let path_wide: Vec<u16> = path
+        .as_ref()
+        .as_os_str()
+        .encode_wide()
+        .chain(Some(0))
+        .collect();
+    let mut volume_name_buffer = vec![0u16; MAX_PATH as usize];
+
+    unsafe { GetVolumePathNameW(PCWSTR(path_wide.as_ptr()), volume_name_buffer.as_mut()) }?;
+    Ok(volume_name_buffer)
+}
+
+/// A wrapper function for
+/// [GetVolumeNameForVolumeMountPointW](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getvolumenameforvolumemountpointw)
+/// that retrieves  a volume GUID path for the volume that is associated with the specified volume
+/// mount point (drive letter, volume GUID path, or mounted folder).
+fn get_volume_guid_path(volume_path_w: &Vec<u16>) -> io::Result<Vec<u16>> {
+    let mut volume_guid_path = vec![0u16; 50usize];
+    unsafe {
+        GetVolumeNameForVolumeMountPointW(PCWSTR(volume_path_w.as_ptr()), volume_guid_path.as_mut())
```

Comment:
Windows typically provides two variants of a function: ANSI and Wide String. The ANSI version doesn't handle UTF-8, so it's generally best to avoid using it. In this code, the conversion to a wide string happens only once in `get_volume_path`, and then the string is passed around as a `Vec<u16>`, which helps avoid additional allocations. I don't think this will impact performance in any significant way.